datus/utils/sql_utils.py

```
@@ -96,54 +106,21 @@ def parse_metadata(sql: str, dialect: str = DBType.SNOWFLAKE) -> Dict[str, Any]:
     Returns a list of unique table names with original case preserved.
     Filters out CTE (Common Table Expression) tables.
     """
-    dialect = parse_dialect(dialect)
     # Parse the SQL using sqlglot
-    parsed = sqlglot.parse_one(sql, dialect=dialect)
+    read_dialect = parse_read_dialect(dialect)
+    try:
+        parsed = sqlglot.parse_one(sql, read=read_dialect, error_level=sqlglot.ErrorLevel.IGNORE)
+        if parsed is None:
+            return []
+    except Exception as e:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)